scripts/qa_crowbarsetup.sh

```diff
@@ -3069,22 +3089,34 @@ function onadmin_rebootcloud()
     pre_hook $FUNCNAME
     get_novacontroller
 
-    local cmachines=$(get_all_discovered_nodes)
+    local cmachines
     local m
+
+    if [[ $hacloud = 1 ]] ; then
+        cluster_node_assignment
+
+        for c in data network services; do
+            local clusternodes_var=$(echo clusternodes${clusternameservices})
+            for m in ${!clusternodes_var}; do
+                power_cycle_and_wait $m
+                m_hostname=$(echo $m | cut -d '.' -f 1)
+                wait_for 400 5 "crowbar node_state status | grep $m_hostname | grep -qiE \"ready$|problem$\"" "node $m_hostname to be online"
+            done
+            complain_if_problem
+        done
+
+        cmachines=$unclustered_nodes
```

Comment:
But `cmachines` is not compute nodes. It's just "nodes that are not in any cluster" (and in the non-HA case, that is "all nodes").

Also, there's no code here to calculate the value of `cmachines`; this is done in `cluster_node_assignment`. So I'm a bit confused by your comment :-)